Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,263 @@
+    """
+    Write updated data to csv file using csv lib
+    :param csv_path: str, path to the folder where the csv file is located
+    :param csv_data: list, data which has to be written to the csv file
+    """
+    with open(csv_path, 'w', newline='', encoding='utf-8') as csv_file:
+        csv_writer = csv.DictWriter(csv_file, fieldnames=csv_data[0].keys())
+        csv_writer.writeheader()
+        csv_writer.writerows(csv_data)
+
+
+def rearrange_data(old_csv_data):
+    """
+    Rearrange data into format 'decades : {countries : [users_data]}'
+    :param old_csv_data: list, data from the csv file in the old format
+    :return: dict, data which was rearranged from list format to the dict format
+    """
+    new_data = {}
+    for row in old_csv_data:
+        decade = row['dob.date'][-2] + '0-th'
+        country = row['location.country']
+        if decade not in new_data:
+            new_data[decade] = {}
+        if country not in new_data[decade]:
+            new_data[decade][country] = []
+        new_data[decade][country].append(row)
+    return new_data
+
+
+def create_sub_folders(rearranged_data):
+    """
+    Create folders with csv files based on the new data format
+    :param rearranged_data: dict, data on the structure of which
+                            folders are created and csv files into folders are filled
+    """
+    for decade in rearranged_data:
+        decade_path = os.path.join(os.getcwd(), decade)
+        os.makedirs(decade_path)
+        for country in rearranged_data[decade]:
+            country_path = os.path.join(decade_path, country)
+            os.makedirs(country_path)
+            users_age = [int(user["dob.age"]) for user in rearranged_data[decade][country]]
+            users_registered_age = [int(user["registered.age"]) for user in rearranged_data[decade][country]]
+            users_id_name = [user["id.name"] for user in rearranged_data[decade][country]]
+            csv_file_name = f'{max(users_age)}_' \
+                            f'{sum(users_registered_age)/len(users_registered_age)}_' \
+                            f'{Counter(users_id_name).most_common(1)[0][0]}'
+            csv_file_path = os.path.join(country_path, csv_file_name)
+            with open(csv_file_path + '.csv', 'w', newline='', encoding='utf-8') as csv_file:
+                csv_writer = csv.DictWriter(csv_file, rearranged_data[decade][country][0].keys())
+                csv_writer.writeheader()
+                csv_writer.writerows(rearranged_data[decade][country])
+
+
+if __name__ == "__main__":
+    # PARSE COMMAND-LINE ARGUMENTS
+    parser = argp.ArgumentParser()
+
+    parser.add_argument('--destination_folder', help='path to the folder where the output file will be placed',
+                        required=True)
+    parser.add_argument('--filename', default='output', help='name of the output file')
+    parser.add_argument('--filtering', help='data filtering method')
```

Comment:
AAAAAAAAAAAAAAAAAAAAAAAAAA.
got it.
this is your way of restriction.
Unfortunately, it's not efficient.
try **mutually_exclusive_group**